src/persist/src/indexed/cache.rs

```diff
@@ -300,6 +305,7 @@ impl<B: Blob> BlobCache<B> {
         let meta = ProtoMeta::decode(&bytes).map_err(|err| {
             Error::from(format!("invalid meta at key {}: {}", Self::META_KEY, err))
         })?;
+        self.check_meta_build_version(&meta)?;
```

Comment:
This will be the "real" backward/forward compatibility logic, so it belongs in persist. For example, if we migrated the way we store some field, this would be a great spot to read the old and new proto fields and use whichever of the two is set to hydrate the in-mem BlobMeta.

The encoding version stuff IMO is basically just automating version upgrade instructions to the user of "while persist is experimental, if you get this error, wipe your data dir". That very well could live inside persist (IIRC this was your preference) but to me it feels like an "external to persist" thing.

Summary: This one def only belongs here, encoding version could be in either place and is a judgement call